RoboCat/Src/Main.cpp

```diff
@@ -21,33 +98,243 @@ int main(int argc, const char** argv)
+					connSocket->Send(msg.c_str(), msg.length());
+				}
+			}
+			listenSocket->CleanUp();
+			return;
 				  });
 
-	while (true)
+	std::thread t2([&win, &quit]()
+		{
+			// While running
+			while (!quit)
+			{
+				std::this_thread::sleep_for(std::chrono::milliseconds(40)); // Without this sleep, the thread may occasionally begin attempting Receive on a VERY fresh socket(), 
+				for (int i = 0; i < connections.size(); i++)				//^throwing an error. The documentation specifies that "some time must pass" between an accept and Receive().
+				{
+					if (connections[i].socketPtr != nullptr)
+					{
+						// Write messages from each connection
+						char buffer[4096];
+						int32_t bytesReceived = connections[i].socketPtr->Receive(buffer, 4096);
+						if (bytesReceived > 0)
+						{
+							std::string receivedMsg(buffer, bytesReceived);
+							char* inputC = new char[receivedMsg.length() + 1];
+							strcpy(inputC, receivedMsg.c_str());
+							char delim[] = " ";
+							char* ptr = strtok(inputC, delim);
+							while (ptr != NULL)
+							{
+								if (ptr != NULL && strcmp(ptr, "/CHANGEALIAS") == 0)
+								{
+									ptr = strtok(NULL, delim);
+									std::string newAlias(ptr);
+									connections[i].alias = newAlias;
+								}
+								ptr = NULL;
+							}
+							if (connections[i].alias.length() > 0)
+							{
+								// Write message
+								LOG("%d - %s: %s", connections[i].connectionID, connections[i].alias.c_str(), receivedMsg.c_str());
+							}
+							else 
+							{
+								// Write Message with no alias
+								LOG("%d - Unkown: %s", connections[i].connectionID, receivedMsg.c_str());
+							}
+						}
+						else if (bytesReceived == 0 || SocketUtil::GetLastError() == 10054)
+						{
+							// Connection closed, forecfully or otherwise - discard.
+							LOG("-------------%s has disconnected-------------", connections[i].alias.c_str());
+							connections[i].socketPtr->CleanUp();
+							connections.erase(connections.begin() + i);
+						}
+					}
+				}
+			}
+			return;
+		});
+
+	while (!quit)
 	{
 		std::string input;
 		std::getline(std::cin, input);
-		win.WriteFromStdin(input);
+		//win.WriteFromStdin(input);
+		win.Write(input);
+
+		// Read inputs to check for commands
+		// /connect IPAddr PortNumber - connect to a peer -> Peers do not syndicate other peers: To form a chat "room", peers must connect to each person they want to talk to.
+		// /disconnect ID - close specified connection
+		// /alias - change name
+		// /list - shows all active connections by ID and alias
+		// /tell ID - sends message to specified ID
+		// /exit - disconnect all and close program
+		// IF NO COMMAND is detected -> sends typed message to all connections.
+		char* inputC = new char[input.length() + 1];
+		strcpy(inputC, input.c_str());
+		char delim[] = " ";
+		char* ptr = strtok(inputC, delim);
+		while (ptr != NULL)
+		{
+			if (strcmp(ptr, "/connect") == 0)
```

Comment:
No `ptr != NULL` in this if check. Not sure you needed it in the other ones -- your while loop checks for it anyway. Just a little inconsistent.